tests/ui/methods/method-self-arg-trait.rs

```diff
@@ -13,32 +13,40 @@ trait Bar : Sized {
 
     fn bar1(&self) {
         unsafe { COUNT *= 7; }
+        //~^ WARN creating a mutable reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.